block/manager.go

```diff
@@ -549,17 +543,13 @@ func (m *Manager) getCommit(header types.Header) (*types.Commit, error) {
 func (m *Manager) IsProposer() (bool, error) {
 	m.lastStateMtx.RLock()
 	defer m.lastStateMtx.RUnlock()
-	// if proposer is not set, assume self proposer
-	if m.lastState.Validators.Proposer == nil {
-		return true, nil
-	}
 
 	signerPubBytes, err := m.proposerKey.GetPublic().Raw()
 	if err != nil {
 		return false, err
 	}
 
-	return bytes.Equal(m.lastState.Validators.Proposer.PubKey.Bytes(), signerPubBytes), nil
+	return bytes.Equal(m.genesis.Validators[0].PubKey.Bytes(), signerPubBytes), nil
```

Comment:
Agreed, we should check for this to avoid seg faults anyway even if we're checking for it when initializing the block manager. That said, is it checked when initializing block manager?